bgpd/bgp_aspath.c

```diff
@@ -1187,6 +1187,33 @@ int aspath_loop_check(struct aspath *aspath, as_t asno)
 	return count;
 }
 
+/* AS path loop check.  If aspath contains asno 
+   that is a confed id then return >= 1. */
+int aspath_loop_check_confed(struct aspath *aspath, as_t asno)
+{
+	struct assegment *seg;
+	int count = 0;
+
+	if ((aspath == NULL) || (aspath->segments == NULL))
+		return 0;
+
+	seg = aspath->segments;
+
+	while (seg) {
+		int i;
+
+		for (i = 0; i < seg->length; i++)
+			if (seg->type != AS_CONFED_SEQUENCE &&
```

Comment:
Are we checking this case from RFC?

```
A BGP speaker receiving an AS_PATH attribute containing an autonomous
   system matching its own AS Confederation Identifier SHALL treat the
   path in the same fashion as if it had received a path containing its
   own AS number.
```

Reply:
yes this function is used to identify the confederation id in the AS_PATH.
since the  the RFC states:

AS Confederation Identifier:
      An externally visible autonomous system number that identifies a
      BGP confederation as a whole.

and if the path segment is  AS_CONFED_SET or  AS_CONFED_SEQUENCE, its an internal to the confederation.